llvm/lib/Transforms/InstCombine/InstCombineSimplifyDemanded.cpp

```diff
@@ -143,7 +155,7 @@ Value *InstCombinerImpl::SimplifyDemandedUseBits(Value *V, APInt DemandedMask,
     return SimplifyMultipleUseDemandedBits(I, DemandedMask, Known, Depth, CxtI);
 
   KnownBits LHSKnown(BitWidth), RHSKnown(BitWidth);
-
+  bool MaySimplifyAsConstant = true;
```

Comment:
Possibly this flag can be replaced by a isPointerTy check below? This seems like it should also hold for other instructions working on pointers, if we were to handle them.